src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -135,7 +135,7 @@ public async Task<SyncTriggersResult> TrySyncTriggersAsync(bool isBackgroundSync
 
                 bool shouldSyncTriggers = true;
                 string newHash = null;
-                if (isBackgroundSync && !_environment.IsKubernetesManagedHosting())
+                if (isBackgroundSync && hashBlobClient != null && !_environment.IsKubernetesManagedHosting())
```

Comment:
As noted in the other open discussion thread, I agree with you that changing this to `if (isBackgroundSync && hashBlobClient != null)` would be best.